src/backend/aspen/covidhub_import/implementation.py

```diff
@@ -120,68 +152,196 @@ def import_project(
+                joinedload(ConsensusGenome.sample_fastqs).joinedload(
+                    SampleFastqs.czb_id
+                )
             )
             .all()
-        )
-        czbid_without_consensus_genomes: Iterable[DphCZBID] = (
-            covidhub_session.query(DphCZBID)
+        }
+
+        internal_czb_ids_metadata: Mapping[str, Tuple[str, datetime.datetime]] = {
+            czbid: (external_accession, collection_date)
+            for czbid, external_accession, collection_date in covidhub_session.query(
+                InternalCZBID.czb_id.label("czb_id"),
+                AccessionSample.accession.label("external_accession"),
+                QPCRRun.completed_at.label("collection_date"),
+            )
             .join(Project)
-            .filter(
-                DphCZBID.czb_id.notin_(
-                    {
-                        consensus_genome.sample_fastqs.czb_id.czb_id
-                        for consensus_genome in consensus_genomes
-                    }
-                )
+            .join(CZBIDRnaPlate)
+            .join(
+                RNAPlate,
+                Extraction,
+                SamplePlate,
+                AccessionSample,
+                Aliquoting,
+                QPCRPlate,
+                QPCRRun,
             )
             .filter(Project.rr_project_id == rr_project_id)
-            .options(
-                joinedload(DphCZBID.sample_fastqs).joinedload(
-                    SampleFastqs.consensus_genomes
-                ),
-                joinedload(DphCZBID.genome_submission_info),
+            .filter(QPCRResult.well_id == CZBIDRnaPlate.well_id)
+            .filter(AccessionSample.well_id == CZBIDRnaPlate.well_id)
+            .group_by(
+                Project.rr_project_id.label("project_id"),
+                Project.collaborating_institution.label("project_name"),
+                AccessionSample.accession.label("external_accession"),
+                InternalCZBID.czb_id.label("czb_id"),
+                QPCRRun.completed_at.label("collection_date"),
             )
-        )
-
-        external_accessions = {
-            consensus_genome.sample_fastqs.czb_id.external_accession
-            for consensus_genome in tqdm.tqdm(consensus_genomes)
-            if isinstance(consensus_genome.sample_fastqs.czb_id, DphCZBID)
         }
+
+        # verify that we have no duplicates for external accessions
+        external_accession_to_czbids: DefaultDict[
+            str, set[str]
+        ] = collections.defaultdict(set)
+        for czbid, (external_accession, _) in internal_czb_ids_metadata.items():
+            external_accession_to_czbids[external_accession].add(czbid)
+        czbids_to_process: MutableSequence[CZBID] = list()
+        for external_accession, czbids in external_accession_to_czbids.items():
+            if len(czbids) > 1:
+                # gather up the information for each czbid representing the external
```

Comment:
So I'm not sure we want this logic. If a county accidentally sent two samples with the same external_accession they may actually be two separate samples. Ideally there wouldn't be any of these, but are there?

Reply:
Yes, there are.

These accessions are assigned when czb got the tubes (`AccessionSamples`), and not assigned by the counties afaik.

Reply:
Example for Marin:

```
External accession H12313 represented by multiple czbids: {'RR089i_00186', 'RR089i_00181'}.  Choosing RR089i_00181 as the czbid forthis sample.
External accession H11851 represented by multiple czbids: {'RR089i_00182', 'RR089i_00187'}.  Choosing RR089i_00182 as the czbid forthis sample.
```

Reply:
Oh interesting these are internal samples that were picked twice, ok yea then this makes sense